portal/src/main/scala/cromwell/pipeline/controller/ProjectController.scala

```diff
@@ -0,0 +1,39 @@
+package cromwell.pipeline.controller
+
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.server.Route
+import akka.http.scaladsl.server.Directives._
+import cromwell.pipeline.datastorage.dto.UserId
+import cromwell.pipeline.datastorage.dto.project.ProjectAdditionRequest
+import cromwell.pipeline.service.ProjectService
+import cromwell.pipeline.utils.auth.AccessTokenContent
+import de.heikoseeberger.akkahttpplayjson.PlayJsonSupport._
+
+import scala.concurrent.ExecutionContext
+import scala.util.{ Failure, Success }
+
+class ProjectController(projectService: ProjectService)(
+  implicit executionContext: ExecutionContext
+) {
+  val route: AccessTokenContent => Route = accessToken =>
+    path("projects") {
+      concat(
+        get {
+          parameter('name.as[String]) { name =>
+            onComplete(projectService.getProjectByName(name, UserId(accessToken.userId))) {
+              case Success(project) => complete(project)
+              case Failure(e)       => complete(StatusCodes.NotFound, e.getMessage)
```

Comment:
You must take into account different cases why you failed to provide an appropriate error code.
1) You could have failed if the access is denied (403).
2) You could have failed if the project not found (404).
3) It is also possible that some unexpected things happened, in which case you should return `interval server error` (500).
If you will provide different exceptions in the service layer for situations 1 and 2, then you will be able to distinguish all of them.

Reply:
Fixed